sinks/logzio/metric_test.go

```diff
@@ -231,3 +235,49 @@ func TestCreateLogzioEvent(t *testing.T) {
 		require.Equal(t, testSample.Tags[key], logzEvent[eventKey])
 	}
 }
+
+func TestLogzioSender(t *testing.T) {
+	t.Parallel()
+	var recordedRequests []byte
+	logger := logrus.New()
+
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		recordedRequests, _ = ioutil.ReadAll(req.Body)
+		rw.WriteHeader(http.StatusOK)
+	}))
+
+	require.NotNil(t, server)
+	defer server.Close()
+
+	logzSink, err := NewLogzioMetricSink(testLogzioToken, testLogzioToken, "", server.URL, logger, dimensions)
+
+	require.NotNil(t, logzSink)
+	require.NoError(t, err)
+
+	testMetric := samplers.InterMetric{
+		Name:      "test.metric",
+		Type:      samplers.StatusMetric,
+		Message:   "test",
+		Timestamp: 1604487565,
+		Value:     123,
+		Tags: []string{
+			"tag:val",
+			"foo:bar",
+		},
+	}
+
+	err = logzSink.Flush(context.TODO(), []samplers.InterMetric{testMetric})
+	assert.NoError(t, err)
+
+	requests := strings.Split(string(recordedRequests), "\n")
+	var logzMetric LogzioMetric
```

Comment:
Maybe assert the recorded requests count? so requests[0] won't cause an error in case it's empty

Reply:
Done